dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx

```diff
@@ -222,20 +222,10 @@ export const ConfigPanel = ({ visualizations, setCurVisId, callback, changeIsVal
     );
   };
 
-  const isLineOrScatter = (vs: string) => (vs === visChartTypes.Line || vs === visChartTypes.Scatter);
-  const isVerticalorHorizontalBar = (vs: string) => (vs === visChartTypes.Bar || vs === visChartTypes.HorizontalBar);
-
   const memorizedVisualizationTypes = useMemo(() => {
     let visDefinition = {}
     return ENABLED_VIS_TYPES.map((vs: string) => {
-      if (isLineOrScatter(vs)) {
-        visDefinition = vs === visChartTypes.Line ? getVisType(vs, { type: visChartTypes.Line }) : getVisType(vs, { type: visChartTypes.Scatter });
-      } else if (isVerticalorHorizontalBar(vs)) {
-        visDefinition = vs === visChartTypes.Bar ? getVisType(vs, { type: visChartTypes.Bar }) : getVisType(vs, { type: visChartTypes.HorizontalBar });
-      } else {
-        visDefinition = getVisType(vs);
-      }
-
+      visDefinition = getVisTypeData(vs);
       return {
         ...visDefinition,
```

Comment:
I do not get it about the spread operator. Why do you have to use so much of it? Your function `getVisTypeData` calls some `getVisType` function and spreads the object to create a new object. Now you get the data into `visDefinition` and create another object by spreading that object. Aren't we overusing it?

Function purity is defined as - The function should not update any variables that are present outside the function or that are passed as variables. 
The variables it has created, what is the point in creating another variable to return it? What you are really doing here is 
```JS
visDefinition = getVisTypeData(vs);
const temp = {...visDefinition}; // you are just not seeing this variable, i.e. you had not explicitly defined it. But the computer has to do it.
return temp;
```
As your target function is returning a renewed object, you can simply return that. Or make your target function to return whatever it has received and create a copy when you update it.

Reply:
Optimized above block of code as per your suggestions.